Panic on Linux/arm64 when job fails

I built the runner from main and ran into a panic when a job fails.
Here's some hard- and software information:
* Broadcom BCM2711 CPU (arm64)
* Raspbian (Debian 11)
* Go 1.20.7

The stacktrace is as follows:
```
panic: runtime error: invalid memory address or nil pointer dereference                                                                           
[signal SIGSEGV: segmentation violation code=0x1 addr=0xd0 pc=0x612e10]                                                                           
                                                                                                                                                  
goroutine 111 [running]:                                                 
github.com/ChristopherHX/github-act-runner/actionsdotnetactcompat.(*ghaFormatter).Format(0x40003fe7e0, 0x40000e9ea0)                              
        /home/fux/Desktop/github-act-runner/actionsdotnetactcompat/act_worker.go:103 +0x420                                                       
github.com/nektos/act/pkg/runner.(*maskedFormatter).Format(0x4000534960, 0x303790?)                                                               
        /home/fux/go/pkg/mod/github.com/christopher!h!x/act@v0.2.23-0.20230807182133-72d04de87e9c/pkg/runner/logger.go:175 +0x4c                  
github.com/sirupsen/logrus.(*Entry).write(0x40000e9ea0)                  
        /home/fux/go/pkg/mod/github.com/sirupsen/logrus@v1.9.0/entry.go:289 +0xd8                                                                 
github.com/sirupsen/logrus.(*Entry).log(0x4000125030, 0x4, {0x40000fe620, 0x67})                                                                  
        /home/fux/go/pkg/mod/github.com/sirupsen/logrus@v1.9.0/entry.go:252 +0x468                                                                
github.com/sirupsen/logrus.(*Entry).Log(0x4000125030, 0x4, {0x4000317cf8?, 0x1?, 0x1?})                                                           
        /home/fux/go/pkg/mod/github.com/sirupsen/logrus@v1.9.0/entry.go:304 +0x60
github.com/sirupsen/logrus.(*Entry).Logf(0x4000125030, 0x4, {0x7d98fc?, 0x4000089800?}, {0x4000317d58?, 0x0?, 0x4000017d78?})                     
        /home/fux/go/pkg/mod/github.com/sirupsen/logrus@v1.9.0/entry.go:349 +0x88                                                                 
github.com/sirupsen/logrus.(*Entry).Infof(...)                                                                                                    
        /home/fux/go/pkg/mod/github.com/sirupsen/logrus@v1.9.0/entry.go:362                                                                       
github.com/nektos/act/pkg/runner.useStepLogger.func1.1({0x40000fe460?, 0x67?})                                                                    
        /home/fux/go/pkg/mod/github.com/christopher!h!x/act@v0.2.23-0.20230807182133-72d04de87e9c/pkg/runner/job_executor.go:181 +0x70            
github.com/nektos/act/pkg/common.(*lineWriter).handleLine(...)           
        /home/fux/go/pkg/mod/github.com/christopher!h!x/act@v0.2.23-0.20230807182133-72d04de87e9c/pkg/common/line_writer.go:45                    
github.com/nektos/act/pkg/common.(*lineWriter).Write(0x400041c880, {0x40004c2000?, 0x925d90?, 0x8000?})                                           
        /home/fux/go/pkg/mod/github.com/christopher!h!x/act@v0.2.23-0.20230807182133-72d04de87e9c/pkg/common/line_writer.go:31 +0x1a8             
github.com/nektos/act/pkg/container.(*ptyWriter).Write(0x40003fce70, {0x40004c2000, 0x7ec, 0x8000})                                               
        /home/fux/go/pkg/mod/github.com/christopher!h!x/act@v0.2.23-0.20230807182133-72d04de87e9c/pkg/container/host_environment.go:174 +0x164    
io.copyBuffer({0x92c620, 0x40003fce70}, {0x92cfe0, 0x4000010220}, {0x0, 0x0, 0x0})                                                                
        /snap/go/current/src/io/io.go:429 +0x1a8                         
io.Copy(...)                                                             
        /snap/go/current/src/io/io.go:386                                                                                                         
github.com/nektos/act/pkg/container.copyPtyOutput({0x92c620?, 0x40003fce70?}, {0x92cfe0?, 0x4000010220?}, 0x400043ec00?)                          
        /home/fux/go/pkg/mod/github.com/christopher!h!x/act@v0.2.23-0.20230807182133-72d04de87e9c/pkg/container/host_environment.go:238 +0x64     
created by github.com/nektos/act/pkg/container.(*HostEnvironment).exec   
        /home/fux/go/pkg/mod/github.com/christopher!h!x/act@v0.2.23-0.20230807182133-72d04de87e9c/pkg/container/host_environment.go:302 +0x4dc
```

Thanks in advance for taking some time to look at this issue :)

--- a/actionsdotnetactcompat/act_worker.go
+++ b/actionsdotnetactcompat/act_worker.go
@@ -94,13 +94,13 @@ func (f *ghaFormatter) Format(entry *logrus.Entry) ([]byte, error) {
 		}
 		stepID = prefix + stepIDArray[0]
 	}
-	cur := f.logger.Current()
-	if !f.main && hasStepID {
+	
+	if cur := f.logger.Current(); cur != nil && !f.main && hasStepID {
 		f.main = true
 		flushInternal(cur, &model.StepResult{Conclusion: model.StepStatusSuccess})
 	}
 	stepName, hasStepName := entry.Data["step"]
-	if hasStepName && hasStepID && f.logger.Current().RefName != stepID {
+	if cur := f.logger.Current(); hasStepName && hasStepID && cur != nil && cur.RefName != stepID {
 		if stage == "Post" {
 			f.Flush()
 		} else if f.result != nil {
@@ -113,7 +113,9 @@ func (f *ghaFormatter) Format(entry *logrus.Entry) ([]byte, error) {
 			te := protocol.CreateTimelineEntry(f.logger.TimelineRecords.Value[0].ID, stepID, stage.(string)+" "+stepName.(string))
 			te.Order = f.logger.TimelineRecords.Value[f.logger.CurrentRecord-1].Order + 1
 			f.logger.Insert(te)
-			f.logger.Current().Start()
+			if cur := f.logger.Current(); cur != nil {
+				cur.Start()
+			}
 			f.logger.Update()
 		} else {
 			for {
@@ -129,22 +131,6 @@ func (f *ghaFormatter) Format(entry *logrus.Entry) ([]byte, error) {
 			f.logger.Update()
 		}
 	}
-	//js, _ := json.Marshal(entry.Data)
-	//entry.Message = (string)(js) + "|" + entry.Message
-
-	// if f.rc != nil && f.rc.Parent == nil && (f.logger.Current() == nil || f.logger.Current().RefName != f.rc.CurrentStep) {
-	// 	if res, ok := f.rc.StepResults[f.logger.Current().RefName]; ok && f.logger.Current() != nil {
-	// 		f.flushInternal(res)
-	// 		for {
-	// 			next := f.logger.MoveNext()
-	// 			if next == nil || next.RefName == f.rc.CurrentStep {
-	// 				break
-	// 			}
-	// 			f.logger.Current().Complete("Skipped")
-	// 		}
-	// 		f.logger.Current().Start()
-	// 	}
-	// }
 	if f.rqt.MaskHints != nil {
 		for _, v := range f.rqt.MaskHints {
 			if strings.ToLower(v.Type) == "regex" {

